docs/semgrep-app/editor.md

```diff
@@ -0,0 +1,205 @@
+
+1. Enter related terms into the search bar to find potential rules. In our example, this is `SHA1.`
+![Screenshot of Library pane with SHA1 filter](../img/editor-SHA1.png)
+
+2. Registry rules cannot be edited directly. They can be **forked** by **right-clicking on their entry** in the Library and selecting **Fork rule.**
+![Screenshot of forking menu](../img/editor-forking.png)
+3. The rule is copied to your organization’s rules.
+4. Enter edits to finalize your rule. In this example, the patterns are changed to find matches for [MD2](https://docs.oracle.com/javase/9/docs/specs/security/standard-names.html#messagedigest-algorithms) and the severity is increased from `WARNING` to `ERROR`.
+5. Enter updates to test cases.
+6. Click **Run** to validate your rule.
+7. Click **Save** to save your rule. The following rule displays the end result.
+
+<iframe title="Prevent use of MD2" src="https://semgrep.dev/embed/editor?snippet=s-santillan:use-of-md2" width="100%" height="432px" frameBorder="0"></iframe>
+
+### Debugging syntax issues
+
+The editor will not save, share, or run if the YAML syntax is **unparseable**. The editor supports error handling, such as for the following cases:
+
+- Invalid language keys
+- Incorrect severity fields
+- Missing `id` key
+- Missing required schema fields
+
+![Screenshot of Editor's error message](../img/editor-errorwarn-message.png)
+
+If no schema or spelling errors are detected, check for indentation or newline issues. You can also get rule-writing help through the [community Slack](https://r2c.dev/slack).
+
+## Evaluating a rule’s performance
+
+Evaluating a rule through testing ensures that it meets your standards for precision and speed, thus **reducing false positives**. Semgrep uses certain annotations in comments to enable users to test which blocks of code are intended to match with a rule, and which blocks of code should not match.
+
+In the following example, run the rule and click on the three-dot menu to view the result of the tests.
+
+<iframe title="Testing example" src="https://semgrep.dev/embed/editor?snippet=s-santillan:snippet=7n1W" width="100%" height="432px" frameBorder="0"></iframe>
+
+To **test a rule**:
+
+1. Create at least one **true positive**: a code sample intended to match the rule.
+2. Above this potential match, create a comment, followed by a space (` `), followed by `ruleid:RULE_ID` which specifies the rule that should match. In the preceding example, this is `//ruleid:hardcoded-conditional`.
+3. Create at least one **true negative**: a code sample intended not to match the rule.
+4. Above this non-match, create a comment followed by a space ( ), followed by `ok:RULE_ID`.
```

Comment:
```suggestion
4. Above this non-match, create a comment followed by a space (` `), followed by `ok:RULE_ID`.
```

Reply:
Have addressed this by writing out the example in full.